src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/LinkSpec.groovy

```diff
@@ -651,6 +654,50 @@ class LinkSpec extends HealthCheckSpecification {
         getIsl().srcSwitch.dpId | getIsl().srcPort | getIsl().dstSwitch.dpId | null      | "dst_port"
     }
 
+    @Tidy
+    @Tags(HARDWARE)
+    def "An ISL between noviflow switches doesn't go down when one switch is DEACTIVATED"(){
+        given: "A noviflow switch"
+        def swToDeactivate = topology.activeSwitches.find { it.noviflow }
+
+        when: "Deactivate the src switch"
+        def relatedIsl = topology.getRelatedIsls(swToDeactivate)
+        def nonNoviflowIsl = relatedIsl.find { !it.dstSwitch.noviflow }
+
+        lockKeeper.knockoutSwitch(swToDeactivate)
+        def IsSwDeactivated = true
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getSwitch(swToDeactivate.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+
+        then: "All ISLs are still DISCOVERED because the system uses round trip latency for ISL alive confirmation"
+        sleep((discoveryTimeout + WAIT_OFFSET / 2) * 1000 as long)
+        // make sure that ISLs are not in FAILED status after timeout
+        Wrappers.timedLoop(WAIT_OFFSET / 2) {
+            withPool {
+                (relatedIsl - nonNoviflowIsl).everyParallel { link ->
+                    assert northbound.getLink(link).state == IslChangeType.DISCOVERED
+                }
+            }
+        }
+
+        and: "All ISLs where at least one switch is not Noviflow are FAILED"
+        withPool {
+            nonNoviflowIsl.everyParallel { link ->
+                assert northbound.getLink(link).state == IslChangeType.FAILED
+            }
+        }
+
+        cleanup:
+        IsSwDeactivated && lockKeeper.reviveSwitch(swToDeactivate)
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert northbound.getSwitch(swToDeactivate.dpId).state == SwitchChangeType.ACTIVATED
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.DISCOVERED
+            }.size() == topology.islsForActiveSwitches.size() * 2
+        }
+    }
+
```

Comment:
I'm surprised that no other tests required any refactoring. There should be tests that check that ISLs go to failed status when a switch fails. And they should become broken now

Reply:
> I'm surprised that no other tests required any refactoring.

I haven't investigated it yet

> There should be tests that check that ISLs go to failed status when a switch fails. And they should become 
broken now

agree